src/content_main.js

```diff
@@ -171,42 +162,134 @@ async function storeRecentTimeSpent(recentTimeSpent) {
+ * getPaymentPointerFromPage gets the payment pointer as a string from the monetization meta tag
+ * on the current page. If there is no monetization meta tag on the page, null is returned.
+ * For more info on the monetization meta tag: https://webmonetization.org/docs/getting-started
+ *
+ * @return {string|null} The payment pointer on the current page, or null if no meta tag is present.
  */
-async function getAndValidatePaymentPointer() {
-	const monetizationMeta = document.querySelector('meta[name="monetization"]');
-	let paymentPointer = (monetizationMeta) ? monetizationMeta.content : null;
-	let isValid = false;
+function getPaymentPointerFromPage() {
+	const monetizationMetaTag = document.querySelector('meta[name="monetization"]');
+	if (monetizationMetaTag) {
+		return monetizationMetaTag.content;
+	} else {
+		return null;
+	}
+}
+
+/**
+ * trackPaymentPointer checks repeatedly (every NEW_PAYMENT_POINTER_CHECK_RATE_MS milliseconds)
+ * to see if a new payment pointer is added to the page. If a new payment pointer is added to the
+ * page, handleValidationAndSetExtensionIcon is called to
+ *	- ensure the payment pointer is validated,
+ *	- store the payment pointer in storage,
+ *	- and update the extension icon to the correct image.
+ */
+function trackPaymentPointer() {
+	let cachedPaymentPointer = null;
+
+	setInterval(async () => {
+		const paymentPointerOnPage = getPaymentPointerFromPage();
+
+		const isNewPaymentPointer =
+			(paymentPointerOnPage !== null) &&
+			(paymentPointerOnPage !== cachedPaymentPointer);
+		if (isNewPaymentPointer) {
+			// Note that paymentPointerOnPage may be newly added to the current page and tab,
+			// but it is possible that this payment pointer was recently validated in another tab.
+			// Therefore it is necessary to use handleValidationAndSetExtensionIcon which checks the
+			// browser storage to see if the payment pointer has been recently validated.
+			handleValidationAndSetExtensionIcon(paymentPointerOnPage);
+			cachedPaymentPointer = paymentPointerOnPage;
+		}
+	}, NEW_PAYMENT_POINTER_CHECK_RATE_MS);
+}
 
-	if (null === monetizationMeta) {
-		/* No monetization meta tag provided */
+/**
+ * handleUpdatingExtensionIconOnTabVisibilityChange ensures that the extension icon is updated
+ * to the correct state when switching between tabs.
+ */
+function handleUpdatingExtensionIconOnTabVisibilityChange() {
+	document.addEventListener('visibilitychange', (event) => {
+		const tabIsVisible = !document.hidden;
+		if (tabIsVisible) {
+			const paymentPointerOnPage = getPaymentPointerFromPage();
+			if (paymentPointerOnPage !== null) {
+				handleValidationAndSetExtensionIcon(paymentPointerOnPage);
+			}
+		}
+	});
+}
+
+/**
+ * Takes a payment pointer and validates it if it has not been validated recently. If the payment
+ * pointer is sucessfuly validated then the extension icon is set to its monetized state, otherwise
+ * it is set to its unmonetized state.
```

Comment:
Edit for readability and typos

> Set the extension icon to its monetized state if the provided payment pointer is valid, otherwise set the extension  icon to its unmonetized state.

Add the additional detail on the "validate if not validated recently" inline with the code.